examples/ReferenceTestV2/reference_consumer_v2.py

```diff
@@ -249,144 +368,189 @@ def on_alert_updates(alerts_by_handle):
+            print(f'State {v.NODETYPE.localname} {v.DescriptorHandle}')
+            operational_state_updates[k].append(v)
+
+    observableproperties.bind(mdib, metrics_by_handle=on_metric_updates)
     observableproperties.bind(mdib, alert_by_handle=on_alert_updates)
     observableproperties.bind(mdib, component_by_handle=on_component_updates)
     observableproperties.bind(mdib, waveform_by_handle=on_waveform_updates)
     observableproperties.bind(mdib, description_modifications=on_description_modification)
+    observableproperties.bind(mdib, operation_by_handle=on_operational_state_updates)
 
-    min_updates = sleep_timer // 5 - 1
-    print('will wait for {} seconds now, expecting at least {} updates per Handle'.format(sleep_timer, min_updates))
+    # now collect reports
+    sleep_timer = 30
+    min_updates = 5
+    print(f'will wait for {sleep_timer} seconds now, expecting at least {min_updates} updates per Handle')
     time.sleep(sleep_timer)
 
+    # now check report count
     step = '4a'
-    info ='count numeric metric state updates'
+    info = 'count numeric metric state updates'
     print(step, info)
     is_ok, result = test_min_updates_per_handle(numeric_metric_updates, min_updates)
-    log_result(is_ok, results, step, info)
+    results_collector.log_result(is_ok, step, info)
 
     step = '4b'
     info = 'count string metric state updates'
     print(step)
-    is_ok, result = test_min_updates_per_handle(string_metric_updates, min_updates,)
-    log_result(is_ok, results, step, info)
+    is_ok, result = test_min_updates_per_handle(string_metric_updates, min_updates)
+    results_collector.log_result(is_ok, step, info)
 
     step = '4c'
     info = 'count alert condition updates'
     print(step)
     is_ok, result = test_min_updates_per_handle(alert_condition_updates, min_updates)
-    log_result(is_ok, results, step, info)
+    results_collector.log_result(is_ok, step, info)
 
     step = '4d'
-    info =' count alert signal updates'
+    info = ' count alert signal updates'
     print(step, info)
     is_ok, result = test_min_updates_per_handle(alert_signal_updates, min_updates)
-    log_result(is_ok, results, step, info)
+    results_collector.log_result(is_ok, step, info)
 
-    step ='4e'
+    step = '4e'
     info = 'count alert system self checks'
     is_ok, result = test_min_updates_per_handle(alert_system_updates, min_updates)
-    log_result(is_ok, results, step, info)
+    results_collector.log_result(is_ok, step, info)
 
     step = '4f'
     info = 'count waveform updates'
+    # 3 waveforms (RealTimeSampleArrayMetric) x 10 messages per second x 100 samples per message
     print(step, info)
     is_ok, result = test_min_updates_per_handle(waveform_updates, min_updates)
-    log_result(is_ok, results, step, info+ ' notifications per second')
+    results_collector.log_result(is_ok, step, info + ' notifications per second')
     if len(waveform_updates) < 3:

```

Comment:
instead of `if` use `results_collector.log_result(len(waveform_updates) >= 3, step, info + ' number of waveforms')`

Reply:
done